src/python_testing/TC_DISHM_3_2.py

```diff
@@ -0,0 +1,170 @@
+        logging.info("OnMode: %s" % (onMode))
+        if onMode == NullValue:
+            return True
+
+        return False
+
+    @async_test_body
+    async def test_TC_DISHM_3_2(self):
+
+        self.endpoint = self.user_params.get("endpoint", 1)
+        logging.info("This test expects to find this cluster on endpoint 1")
+
+        asserts.assert_true(self.check_pics("DISHM.S.A0000"), "DISHM.S.A0000 must be supported")
+        asserts.assert_true(self.check_pics("DISHM.S.A0001"), "DISHM.S.A0001 must be supported")
+        asserts.assert_true(self.check_pics("DISHM.S.C00.Rsp"), "DISHM.S.C00.Rsp must be supported")
+        asserts.assert_true(self.check_pics("DISHM.S.C01.Tx"), "DISHM.S.C01.Tx must be supported")
+
+        if not self.check_pics("DISHM.S.A0002"):
+            logging.info("Test skipped because PICS DISHM.S.A0002 (StartupMode) is not set")
+            return
+
+        depOnOffKey = "DISHM.S.F00"
+        asserts.assert_true(depOnOffKey in self.matter_test_config.pics, "%s must be provided" % (depOnOffKey))
+
+        ret = await self.check_preconditions(self.endpoint)
+        asserts.assert_true(ret, "invalid preconditions - StartUpMode overridden by OnMode")
+        attributes = Clusters.DishwasherMode.Attributes
+
+        from enum import Enum
+
+        class CommonCodes(Enum):
+            SUCCESS = 0x00
+            UNSUPPORTED_MODE = 0x01
+            GENERIC_FAILURE = 0x02
+
+        self.print_step(1, "Commissioning, already done")
+
+        self.print_step(2, "Read StartUpMode attribute")
+
+        startup_mode_dut = await self.read_mode_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.StartUpMode)
+
+        logging.info("StartUpMode: %s" % (startup_mode_dut))
+
+        asserts.assert_true(type_matches(startup_mode_dut, uint) or startup_mode_dut ==
+                            NullValue, "Startup mode value should be an integer value or null")
+
+        if startup_mode_dut == NullValue:
+
+            self.print_step(3, "Read SupportedModes attribute")
+            supported_modes_dut = await self.read_mode_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.SupportedModes)
+
+            logging.info("SupportedModes: %s" % (supported_modes_dut))
+
+            asserts.assert_greater_equal(len(supported_modes_dut), 2, "SupportedModes must have at least two entries!")
+
+            for m in supported_modes_dut:
+                new_start_up_mode_th = m.mode
+                break
+
+            self.print_step(4, "Write to the StartUpMode Attribute")
+            await self.write_start_up_mode(newMode=new_start_up_mode_th)
+
+        else:
+            new_start_up_mode_th = startup_mode_dut
+
+        self.print_step(5, "Read CurrentMode attribute")
+
+        old_current_mode_dut = await self.read_mode_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.CurrentMode)
+        self.default_controller.ExpireSessions(self.dut_node_id)
```

Comment:
Why is this ExpireSessions here?  I don't see it in the test plan....  If the idea is to save time or avoid errors, when the device is power-cycled, shouldn't this be at the point where the power-cycle happens?